app/src/components/Tabs/MyProfileTab/ManagePasswordTabContent.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { FC, useEffect, useState } from "react";
+
+import {
+  Box,
+  Center,
+  ProgressCircle,
+  HStack,
+  Input,
+  Select,
+  Icon,
+  createListCollection,
+  Text,
+  VStack,
+} from "@chakra-ui/react";
+
+import { MdInfoOutline, MdSearch } from "react-icons/md";
+import { TitleMedium } from "@/components/Texts/Title";
+
+import { TFunction } from "i18next";
+import PasswordInput from "@/components/password-input";
+import { Button } from "@/components/ui/button";
+import { logger } from "@/services/logger";
+import { useRouter, useSearchParams } from "next/navigation";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { Field } from "@/components/ui/field";
+import { api } from "@/services/api";
+import { Toaster, toaster } from "@/components/ui/toaster";
+
+interface ManagePasswordProps {
+  t: TFunction;
+}
+
+type Inputs = {
+  currentPassword: string;
+  newPassword: string;
+  confirmPassword: string;
+};
+
+const ManagePasswordTabContent: FC<ManagePasswordProps> = ({ t }) => {
+  const searchParams = useSearchParams();
+  const [error, setError] = useState("");
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors, isSubmitting },
+    setError: setFormError,
+    watch,
+    reset,
+  } = useForm<Inputs>();
+
+  const [updatePassword, { isLoading, isError, isSuccess }] =
+    api.useUpdatePasswordMutation();
+
+  const watchPassword = watch("newPassword", "");
+  const onSubmit: SubmitHandler<Inputs> = async (data) => {
+    if (data.newPassword !== data.confirmPassword) {
+      setFormError("confirmPassword", {
+        type: "custom",
+        message: "Passwords don't match!",
+      });
+      return;
+    }
+    const body = {
+      currentPassword: data.currentPassword,
+      confirmPassword: data.confirmPassword,
```

Comment:
Not sure if we need to send this along as well, it's only a safety feature for the user to ensure they remember the password and haven't mistyped it.